### PR TITLE
EDM-4699: Helm chart support for binding PVCs to pre-provisioned PVs

### PR DESCRIPTION
**Pull request by asafbennatan**
# Release target (required before merge to `main`)

Add at least one **`release-X.Y`** label to indicate which release(s) this change targets:

- [x] **`release-1.3`** — inferred from EDM-4699's Definition of Done ("Release notes updated for 1.3.0"); the `fixVersions` field itself could not be confirmed via Jira (MCP connection dropped mid-session — see `01-context.md`).

## EDM-4699: Helm chart support for binding PVCs to pre-provisioned PVs

**Jira:** https://redhat.atlassian.net/browse/EDM-4699
**Source RFE:** EDMRFE-129

### Summary

Adds optional `volumeName` and `selector.matchLabels` Helm values for the builtin PostgreSQL database PVC and the Alertmanager `StatefulSet`'s PVC, so administrators can bind either to a specific pre-provisioned `PersistentVolume` on clusters without dynamic storage provisioning. Both fields are additive and off by default — existing installs render byte-identical manifests unless the new values are explicitly set. Documents the recommended static-storage install pattern, including prerequisites and manual-workaround references, in the Kubernetes install guide.

### Changes

- **`deploy/helm/flightctl/templates/db/flightctl-db-persistentvolumeclaim.yaml`** — added `volumeName`/`selector.matchLabels` to the database `PersistentVolumeClaim` spec, guarded with the chart's existing `with` idiom (nested guard to stay safe when `selector` itself is `null`).
- **`deploy/helm/flightctl/templates/alertmanager/flightctl-alertmanager.yaml`** — added the same two fields to the `StatefulSet.volumeClaimTemplates[0].spec` block.
- **`deploy/helm/flightctl/values.yaml.gotmpl`** / **`values.schema.json`** — new `db.builtin.volumeName`, `db.builtin.selector.matchLabels`, `alertmanager.volumeName`, `alertmanager.selector.matchLabels` values, with doc comments covering the PVC-immutability caveat (Kubernetes forbids changing `volumeName`/`selector` on an already-created PVC) and the PV-capacity requirement (the PV must be at least as large as the requested storage). `values.yaml` and `README.md` regenerated via `make generate`.
- **`docs/user/installing/installing-service-on-kubernetes.md`** — new "Binding to pre-provisioned PersistentVolumes" section: prerequisites (DB `fsGroup` ownership, avoiding NFS for DB storage, PV capacity for both components), example values, an immutability warning, and references to the existing manual workarounds (PVC pre-creation/adoption, PV `claimRef` pre-binding) for patterns the new values don't cover. Cross-referenced from the OpenShift and MicroShift install sections.
- **`.spelling`** — added technical terms used by the new documentation.

### Testing

- **Unit tests:** N/A — no Go code changed.
- **Integration tests:** N/A — no Go code changed; full suite re-run as a regression safety net (1054 tests, 0 failures).
- **Chart verification:** No automated chart-test framework exists in this repo. Verified manually via `helm template`/`helm lint` across all render-branch combinations (unset; `volumeName` only; `selector.matchLabels` only; both; `selector: null` edge case; unsupported key rejected) for both the DB and Alertmanager PVCs. Confirmed zero-diff rendering when the new fields are unset.
- **Full validation suite:** `make generate`, `make lint-helm`, `make lint-docs`, `make spellcheck-docs`, `make lint`, `make unit-test` (5053 tests), `make integration-test` (1054 tests) — all pass. Full details in `.artifacts/implement/EDM-4699/05-validation-report.md` (not committed — local workflow artifact).
- **External review:** Reviewed with the CodeRabbit CLI; actionable findings addressed (see commit history).

### Acceptance Criteria

- [x] AC-1: Optional `volumeName` on the DB **and** Alertmanager PVCs.
- [x] AC-2: Optional `selector.matchLabels` on the DB **and** Alertmanager PVCs.
- [x] AC-3: Both fields optional and off by default — no change to existing dynamic-provisioning behavior.
- [x] AC-4: `values.schema.json` updated for both components.
- [x] AC-5: Chart `README.md` documents the new values.
- [x] AC-6: Kubernetes install guide documents the pattern, examples, and prerequisites.
- [x] AC-7: `make lint-helm` passes with the new fields present, for both components, set and unset.

### Out of scope (per EDM-4699)

- Dynamic StorageClass provisioning/creation.
- Automated PV/PVC creation outside the Helm chart.
- The KV store (Redis) — confirmed to have no persistent storage PVC.
- QE validation of the pre-provisioned-PV install path and the 1.3.0 release-notes update are Definition-of-Done items for other roles, tracked outside this PR.

### CODE REVIEW

File: deploy/helm/flightctl/templates/db/flightctl-db-persistentvolumeclaim.yaml [hunk 1]
```diff
@@ -19,6 +19,16 @@ spec:
   {{- with .Values.global.storageClassName }}
   storageClassName: {{ . }}
   {{- end }}
+  {{- with .Values.db.builtin.volumeName }}
+  volumeName: {{ . }}
+  {{- end }}
```
[the remark this answers is not in the document]
Comment (asafbennatan) [reply deploy/helm/flightctl/templates/db/flightctl-db-persistentvolumeclaim.yaml]: Confirmed and fixed — verified against the Kubernetes docs, which show this exact pattern ([Persistent Volumes: Reserving a PersistentVolume](https://kubernetes.io/docs/concepts/storage/persistent-volumes/#reserving-a-persistentvolume)): a PVC with `volumeName` set but `storageClassName` omitted can still get a default class injected by the `DefaultStorageClass` admission controller, which breaks the static bind if the target PV doesn't match.

Fixed in a2804a13f for both the DB and Alertmanager PVC templates: when `volumeName` or `selector.matchLabels` is set and `global.storageClassName` is not explicitly configured, we now render `storageClassName: ""` instead of omitting the field. The pure dynamic-provisioning path (neither set) is unchanged. Verified rendering across all field combinations.

File: docs/user/installing/installing-service-on-kubernetes.md [hunk 2]
```diff
@@ -27,6 +27,44 @@ You can install the Flight Control Service on any certified Kubernetes distribut
...
+
+If you need a binding pattern the `volumeName` and `selector.matchLabels` values do not cover, such as `matchExpressions`, or changing the binding on an already-installed release, manage the `PersistentVolumeClaim` outside the values covered here:
+
+- Pre-create the `PersistentVolumeClaim` with the exact specification you need, then add the Helm ownership label and annotations (`app.kubernetes.io/managed-by: Helm`, `meta.helm.sh/release-name`, `meta.helm.sh/release-namespace`) before running `helm install`, so Helm adopts the existing resource instead of creating a new one. See [Using Kubernetes Secrets](configuring-external-database.md#using-kubernetes-secrets) in [Configuring External PostgreSQL Database](configuring-external-database.md) for the equivalent adoption pattern applied to secrets.
+- Pre-bind the `PersistentVolume` to the future `PersistentVolumeClaim` by setting the `PersistentVolume`'s `spec.claimRef` to the claim's name and namespace before the claim exists. Kubernetes then reserves that `PersistentVolume` for the matching claim once it is created.
```
[the remark this answers is not in the document]
Comment (asafbennatan) [reply docs/user/installing/installing-service-on-kubernetes.md]: Good catch — fixed in a2804a13f. Scoped the Helm-ownership-adoption bullet to the database `PersistentVolumeClaim`, since it's the only one Helm renders directly. Added a separate Alertmanager-specific bullet describing its actual adoption path: pre-creating a `PersistentVolumeClaim` named `flightctl-alertmanager-data-flightctl-alertmanager-0` (the name the `StatefulSet` controller expects), which it reuses instead of creating a new one from `volumeClaimTemplates`.

To be clear, this only affects a secondary fallback-workaround paragraph — the main `volumeName`/`selector.matchLabels` values this PR adds fully support Alertmanager already, unaffected by this fix.

File: docs/user/installing/installing-service-on-kubernetes.md [hunk 3]
```diff
@@ -27,6 +27,47 @@ You can install the Flight Control Service on any certified Kubernetes distribut
...
+  #   matchLabels:
+  #     disk: standard
+```
+
+Both values are optional and unset by default, which preserves the existing dynamic-provisioning behavior. Set only one of `volumeName` or `selector.matchLabels` per component to keep the binding unambiguous.
+
+Setting either value on a component also makes Flight Control render `storageClassName: ""` on that `PersistentVolumeClaim` when `global.storageClassName` is not explicitly set, instead of omitting the field. This keeps the binding working regardless of whether the cluster has a default `StorageClass` — without it, Kubernetes could inject a default class that does not match the pre-provisioned `PersistentVolume` and leave the claim unbound.
+
+> [!IMPORTANT]
+> `volumeName` and `selector` take effect only when Flight Control first creates the `PersistentVolumeClaim`. Kubernetes does not allow changing these fields on an existing `PersistentVolumeClaim`, so setting or changing them on an already-installed release causes `helm upgrade` to fail for that resource. Set the values before the first `helm install`.
```
Comment (galel12) [inline docs/user/installing/installing-service-on-kubernetes.md]: nit: for Alertmanager this fails because StatefulSet `volumeClaimTemplates` are immutable, not because the PVC fields themselves changed — same "set before first install" advice either way, just a slightly different failure mode than the DB PVC path

Comment (asafbennatan) [reply docs/user/installing/installing-service-on-kubernetes.md]: Good catch, fixed in afc942152. The note now says Alertmanager's failure comes from Kubernetes forbidding changes to an existing `StatefulSet`'s `volumeClaimTemplates` at all, rather than PVC-field immutability like the database path — same practical guidance either way, just accurate on the mechanism now.

**Review by galel12 — APPROVED** [review]
LGTM
